src/libraries/Common/src/System/Security/Cryptography/CompositeMLDsaManaged.cs

```diff
@@ -0,0 +1,682 @@
+            //      As in FIPS 204, len(ctx) is encoded as a single unsigned byte.
+            //      Randomize the message representative
+            //
+            //          r = Random(32)
+            //          M' :=  Prefix || Domain || len(ctx) || ctx || r
+            //                                              || PH( M )
+
+            Span<byte> r = stackalloc byte[CompositeMLDsaAlgorithm.RandomizerSizeInBytes];
+            RandomNumberGenerator.Fill(r);
+
+            byte[] M_prime = GetMessageRepresentative(AlgorithmDetails, context, r, data);
+
+            //  3.  Separate the private key into component keys
+            //      and re-generate the ML-DSA key from seed.
+            //
+            //          (mldsaSeed, tradSK) = DeserializePrivateKey(sk)
+            //          (_, mldsaSK) = ML-DSA.KeyGen(mldsaSeed)
+
+            /* no-op */
+
+            //  4.  Generate the two component signatures independently by calculating
+            //      the signature over M' according to their algorithm specifications.
+            //
+            //          mldsaSig = ML-DSA.Sign( mldsaSK, M', ctx=Domain )
+            //          tradSig = Trad.Sign( tradSK, M' )
+
+            //  Note that in step 4 above, both component signature processes are
+            //  invoked, and no indication is given about which one failed.This
+            //  SHOULD be done in a timing-invariant way to prevent side-channel
+            //  attackers from learning which component algorithm failed.
+
+            Span<byte> randomizer = destination.Slice(0, CompositeMLDsaAlgorithm.RandomizerSizeInBytes);
+            Span<byte> mldsaSig = destination.Slice(CompositeMLDsaAlgorithm.RandomizerSizeInBytes, AlgorithmDetails.MLDsaAlgorithm.SignatureSizeInBytes);
+            Span<byte> tradSig = destination.Slice(CompositeMLDsaAlgorithm.RandomizerSizeInBytes + AlgorithmDetails.MLDsaAlgorithm.SignatureSizeInBytes);
+
+            bool mldsaSigned = false;
+            bool tradSigned = false;
+
+            try
+            {
+                _mldsa.SignData(M_prime, mldsaSig, AlgorithmDetails.DomainSeparator);
+                mldsaSigned = true;
+            }
+            catch (CryptographicException)
+            {
+            }
+
+            int tradBytesWritten = 0;
+
+            try
+            {
+                tradSigned = _componentAlgorithm.TrySignData(M_prime, tradSig, out tradBytesWritten);
+                Debug.Assert(tradSigned, "Signature buffer should be exactly sized.");
+            }
+            catch (CryptographicException)
+            {
+            }
+
+            //  5.  If either ML-DSA.Sign() or Trad.Sign() return an error, then this
+            //      process MUST return an error.
+            //
+            //          if NOT mldsaSig or NOT tradSig:
+            //              output "Signature generation error"
+
+            if (!mldsaSigned || !tradSigned)
```

Comment:
Technically... this is side-effect timing leaking.

If you want to hide as much as possible, you want one boolean, `sigOK` (or whatever) that each catch sets to false.  Or you want an int that each signature success increments, and here it needs to be 2.  Or you can `|` these together instead of using `||`.